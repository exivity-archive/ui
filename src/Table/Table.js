import PropTypes from 'prop-types'
import { Table } from 'reakit'

import withEnumProps from '../withEnumProps'

Table.propTypes = {
  ...Table.propTypes,
  children: PropTypes.node,

  scale: PropTypes.number,
  xsmall: PropTypes.bool,
  small: PropTypes.bool,
  large: PropTypes.bool,
  xlarge: PropTypes.bool,

  compact: PropTypes.bool
}

Table.defaultProps = {
  scale: 1
}

Table.displayName = 'Table'

export default withEnumProps(Table, { scale: 'value' })
