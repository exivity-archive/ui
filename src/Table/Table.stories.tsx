import { storiesOf } from '@storybook/react'
import React from 'react'

import { Table } from '.'

const Sample = () => <>
  <caption align='bottom'>Component dependencies</caption>
  <thead>
  <tr>
    <th>Component</th>
    <th>Dependency</th>
    <th>Version</th>
  </tr>
  </thead>
  <tbody>
  <tr>
    <td>Code</td>
    <td>react-syntax-highlighter<br/>refractor</td>
    <td>^9.0.0<br/>latest</td>
  </tr>
  <tr>
    <td>Markdown</td>
    <td>react-markdown</td>
    <td>^4.0.3</td>
  </tr>
  </tbody>
</>

storiesOf('atoms|Table', module)
  .add('default', () => <Table><Sample /></Table>)
  .add('middle', () => <Table align='middle'><Sample /></Table>)
  .add('bottom', () => <Table align='bottom'><Sample /></Table>)
  .add('compact', () => <Table compact><Sample /></Table>)
