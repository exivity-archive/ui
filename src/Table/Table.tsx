import styled from 'styled-components'
import { fromTheme, globalFont, StyledProps } from '../utils/styled'
import { preciseEm } from '../utils/styled/isolated'

interface TableProps extends StyledProps {
  compact?: boolean
}

export const Table = styled.table<TableProps>`
  ${globalFont};

  table-layout: fixed;
  border-collapse: collapse;
  line-height: 200%;

  tbody,
  td,
  th,
  tfoot,
  thead,
  tr {

  }

  thead tr,
  tbody tr:not(:last-child) {
    border-bottom: 1px solid ${fromTheme(theme => theme.colors.lightGray)};
  }

  tfoot tr {
    border-top: 1px solid ${fromTheme(theme => theme.colors.lightGray)};
  }

  caption {
    text-transform: uppercase;
    font-size: ${preciseEm(0.9)}em;
    color: ${fromTheme(theme => theme.colors.gray)};
  }

  td,
  th {
    padding: ${props => props.compact ? preciseEm(0) : preciseEm(0.2)}em 0;
    vertical-align: middle;

    &:not(:first-child) {
      padding-left: ${props => props.compact ? preciseEm(0.1) : preciseEm(0.4)}em;
    }

    &:not(:last-child) {
      padding-right: ${props => props.compact ? preciseEm(0.1) : preciseEm(0.4)}em;
    }
  }

  th {
    font-weight: bold;
    text-align: left;
  }
`

Table.displayName = 'Table'
