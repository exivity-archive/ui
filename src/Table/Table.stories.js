import React from 'react'
import { Divider } from 'reakit'
import { storiesOf } from '@storybook/react'

import Code from '../Code'
import Table from './Table'

const SimpleTableContents = () => <React.Fragment>
  <thead>
    <tr>
      <th>Component</th>
      <th>Dependency</th>
      <th>Version</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>CodeBlock</td>
      <td>react-syntax-highlighter</td>
      <td>^9.0.0</td>
    </tr>
    <tr>
      <td>Markdown</td>
      <td>react-markdown</td>
      <td>^4.0.3</td>
    </tr>
  </tbody>
</React.Fragment>

storiesOf('atoms|Table', module)
  .add('default', () => <Table>
    <React.Fragment>
      <caption>Table caption</caption>
      <thead>
        <tr>
          <th>Component</th>
          <th>Dependency</th>
          <th>Version</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td><Code>CodeBlock</Code></td>
          <td>react-syntax-highlighter</td>
          <td>^9.0.0</td>
        </tr>
        <tr>
          <td><Code>Markdown</Code></td>
          <td>react-markdown</td>
          <td>^4.0.3</td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td colSpan={3}>I'm at the bottom because I'm a <Code>tfoot</Code>.</td>
        </tr>
      </tfoot>
    </React.Fragment>
  </Table>)
  .add('sizes', () => <React.Fragment>
    <Table xsmall><SimpleTableContents /></Table>
    <Divider />
    <Table small><SimpleTableContents /></Table>
    <Divider />
    <Table><SimpleTableContents /></Table>
    <Divider />
    <Table large><SimpleTableContents /></Table>
    <Divider />
    <Table xlarge><SimpleTableContents /></Table>
  </React.Fragment>)
  .add('compact', () => <div>
    <Table xsmall compact><SimpleTableContents /></Table>
    <Divider />
    <Table small compact><SimpleTableContents /></Table>
    <Divider />
    <Table compact><SimpleTableContents /></Table>
    <Divider />
    <Table large compact><SimpleTableContents /></Table>
    <Divider />
    <Table xlarge compact><SimpleTableContents /></Table>
  </div>)
