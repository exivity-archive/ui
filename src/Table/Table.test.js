/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import Table from './Table'

test('renders default Table', () => {
  const component = renderer.create(<Table />)
  expect(component.toJSON()).toMatchSnapshot()
})
