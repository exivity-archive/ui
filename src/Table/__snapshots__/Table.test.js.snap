// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Table 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

<table
  className="c0"
  scale={1}
/>
`;
